src/libasr/pass/intrinsic_function_registry.h

```diff
@@ -463,6 +464,94 @@ namespace Abs {
 
 } // namespace Abs
 
+namespace Ord{
+    static ASR::expr_t* eval_Ord(Allocator &al, const Location &loc,
+            Vec<ASR::expr_t*> &args) {
+        LCOMPILERS_ASSERT(ASRUtils::all_args_evaluated(args));
+        ASR::ttype_t *type = ASRUtils::TYPE(ASR::make_Integer_t(al, loc, 4, nullptr, 0));
+        ASR::expr_t *expr = args[0];
+        ASR::ttype_t *t = ASRUtils::expr_type(expr);
+        if (ASRUtils::is_character(*t)) {
```

Comment:
Wait a minute, this seems to be repeated. I think we can compute the value in the create_Ord itself.

Reply:
Yess I followed the norm here, all possible error checking in `create_Ord`  abd then the compile time calculation in `eval_Ord`.